### PR TITLE
Spark: Add tests for identity partition data

### PR DESCRIPTION
**Pull request by rdblue**
This adds missing tests that that identity-partition data values are joined correctly to rows when reading data in Spark. This doesn't test cases where physical columns are missing or partitions based on nested fields.

### CODE REVIEW

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  private final String format;
+
+  public TestIdentityPartitionData(String format) {
+    this.format = format;
+  }
+
+  private static SparkSession spark = null;
+  private static JavaSparkContext sc = null;
```
Comment (openinx) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: nit: looks like the var is assigned but never used ?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        new String[] { "date", "level", "message" },
+        new String[] { "level", "date", "message" },
+        new String[] { "date", "message", "level" },
+        new String[] { "level", "message", "date" },
+        new String[] { "message", "date", "level" },
+        new String[] { "message", "level", "date" }
+    };
+
+    for (String[] ordering : cases) {
+      List<Row> expected = logs.select("id", ordering).orderBy("id").collectAsList();
```
Comment (openinx) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: nit: I think the testFullProjection is a special case of testProjections , would merge them together ?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+@RunWith(Parameterized.class)
+public class TestIdentityPartitionData  {
+  private static final Configuration CONF = new Configuration();
+  private static final HadoopTables TABLES = new HadoopTables(CONF);
+
+  @Parameterized.Parameters
+  public static Object[][] parameters() {
+    return new Object[][] {
+        new Object[] { "parquet" },
+        new Object[] { "avro" }
```
Comment (rdsr) [inline spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: Do you want to include ORC as well?

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java (hunk 2 above)
Comment (rdblue) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: If this test case fails, it's easier to debug and find what happened than the other case. That's why I included it.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java (hunk 3 above)
Comment (rdblue) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: I tried, but ORC writes aren't currently supported in Spark.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java (hunk 1 above)
Comment (rdblue) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: Removed.

File: spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java (hunk 3 above)
Comment (rdsr) [reply spark/src/test/java/org/apache/iceberg/spark/source/TestIdentityPartitionData.java]: oh ok. I'll file a ticket for that.

**Comment by rdblue** [comment]
@rdsr, good to commit this?

**Comment by rdsr** [comment]
+1 . I'll commit this

**Comment by rdblue** [comment]
Thanks for merging and reviewing, @rdsr and @openinx!

**Pull request merged**